services/src/main/java/org/keycloak/services/resources/admin/RealmAdminResource.java

```diff
@@ -439,10 +440,7 @@ public Response updateRealm(final RealmRepresentation rep) {
             adminEvent.operation(OperationType.UPDATE).representation(StripSecretsUtils.strip(rep)).success();
             
             if (rep.isDuplicateEmailsAllowed() != null && rep.isDuplicateEmailsAllowed() != wasDuplicateEmailsAllowed) {
-                LegacySessionSupportProvider legacySessionSupportProvider = session.getProvider(LegacySessionSupportProvider.class);
-                if (legacySessionSupportProvider != null) {
-                    legacySessionSupportProvider.clearUserCache();
-                }
```

Comment:
For `session.invalidate` to work, there must be a corresponding `InvalidationHandler` provider factory.

In this case, `InfinispanUserCacheProviderFactory` should implement it. I will merge this PR which otherwise looks good and update it accordingly before pushing.